### PR TITLE
test: rename integration test

### PR DESCRIPTION
**Pull request by olavloite**
Rename the SessionPoolIntegrationTest to IT... so it is actually picked up by the configuration that executes the integration tests.

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java [hunk 1]
```diff
@@ -97,28 +98,28 @@ public ScheduledExecutorService get() {
...
+      assertThat(session.get()).isNotNull();
     }
 
-    try (Session session = pool.getReadSession()) {
-      assertThat(session).isNotNull();
-      Session session2 = pool.getReadSession();
-      assertThat(session2).isNotNull();
+    try (PooledSessionFuture session = pool.getReadSession()) {
+      assertThat(session.get()).isNotNull();
+      PooledSessionFuture session2 = pool.getReadSession();
```
Comment (thiagotnunes) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java:107]: nit: move `session2` declaration into the `try` with resources block

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java [hunk 2]
```diff
@@ -97,28 +98,28 @@ public ScheduledExecutorService get() {
 
   @Test
   public void sessionCreation() {
-    try (Session session = pool.getReadSession()) {
-      assertThat(session).isNotNull();
+    try (PooledSessionFuture session = pool.getReadSession()) {
+      assertThat(session.get()).isNotNull();
```
Comment (thiagotnunes) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java:102]: WDYT about adding timeouts to the `get` calls for the futures (in order not to block indefinitely)?

Comment (olavloite) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java:102]: I would prefer not do that, as the client library also calls `PooledSessionFuture.get()` without a timeout, and I would prefer to keep these consistent with the behavior of the client library itself. Instead, the test case itself will eventually timeout if the calls should block indefinitely.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java (hunk 1 above)
Comment (olavloite) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java:107]: Done.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java (hunk 2 above)
Comment (thiagotnunes) [reply google-cloud-spanner/src/test/java/com/google/cloud/spanner/ITSessionPoolIntegrationTest.java:102]: Cool, thanks for the explanation!

**Pull request merged**